p2p/src/server.rs

```diff
@@ -0,0 +1,287 @@
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
+
+extern crate rand_os;
+extern crate x25519_dalek;
+
+use avrio_config::config;
+use rand_os::OsRng;
+
+use x25519_dalek::EphemeralSecret;
+use x25519_dalek::PublicKey;
+
+use crate::{
+    io::{read, send},
+    peer::add_peer,
+};
+
+use std::convert::TryInto;
+pub struct P2pServer {
+    state: P2pServerState,
+    connections: u64,
+    pub ip: String,
+    pub port: String,
+    pub accept_in: bool,
+    tx: Option<std::sync::mpsc::Sender<&'static str>>,
+}
+
+impl Default for P2pServer {
+    fn default() -> Self {
+        Self {
+            state: P2pServerState::Uninitialized,
+            connections: 0,
+            ip: "0.0.0.0".to_string(),
+            port: "56789".to_string(),
+            accept_in: true,
+            tx: None,
+        }
+    }
+}
+#[derive(PartialEq, Debug)]
+enum P2pServerState {
+    AcceptingIncoming,
+    DenyNewConnections,
+    ShuttingDown,
+    Uninitialized,
+}
+fn from_slice(bytes: &[u8]) -> [u8; 32] {
+    let mut array = [0; 32];
+    let bytes = &bytes[..array.len()]; // panics if not enough data
+    array.copy_from_slice(bytes);
+    array
+}
+
+impl P2pServer {
+    pub fn new(ip: String, port: String, accept_in: bool) -> P2pServer {
+        return P2pServer {
+            state: P2pServerState::Uninitialized,
+            connections: 0,
+            ip,
+            port,
+            accept_in,
+            tx: None,
+        };
+    }
+    fn set_state(&mut self, state: P2pServerState) {
+        self.state = state;
+    }
+    fn increment_connections(&mut self) {
+        if self.state != P2pServerState::DenyNewConnections
+            && self.state != P2pServerState::Uninitialized
+            && self.state != P2pServerState::ShuttingDown
+        {
+            self.connections += 1;
+        }
+    }
+    fn deincrement_connections(&mut self) {
```

Comment:
Should be `decrement_connections`.

Reply:
Solved, also i need to call those functions